docs/source/howto/query.rst

```diff
@@ -90,6 +94,14 @@ For example, you can iterate over the results of your query in a for loop:
     for entry in qb.iterall():
         # do something with a single entry in the query result
 
+.. important::
+
+    When the query result is expected to be large, do not use ``dict`` and ``all`` but use the generator variants ``iterdict`` and ``iterall``.
+    The ``dict`` and ``all`` methods load the entire query result in memory which can lead to out-of-memory errors.
+    It also makes updating the data very slow as each change results in a single database commit.
+    The ``iterdict`` and ``iterall`` generators open a database transaction, committing all changes only once at the end of the generator.
```

Comment:
Maybe we should add a note that in case e.g. a loop running over the generator gets interrupted, none of the changes are actually committed, or stress this somehow.